packages/apollo-server-defer/src/__tests__/starWarsSchema.ts

```diff
@@ -0,0 +1,337 @@
+ *     id: String!
+ *     name: String
+ *     friends: [Character]
+ *     appearsIn: [Episode]
+ *     secretBackstory: String
+ *     primaryFunction: String
+ *   }
+ */
+const droidType = new GraphQLObjectType({
+  name: 'Droid',
+  description: 'A mechanical creature in the Star Wars universe.',
+  fields: () => ({
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: 'The id of the droid.',
+    },
+    name: {
+      type: GraphQLString,
+      description: 'The name of the droid.',
+    },
+    friends: {
+      type: new GraphQLList(characterInterface),
+      description:
+        'The friends of the droid, or an empty list if they have none.',
+      resolve: droid => getFriends(droid),
+    },
+    appearsIn: {
+      type: new GraphQLList(episodeEnum),
+      description: 'Which movies they appear in.',
+    },
+    secretBackstory: {
+      type: GraphQLString,
+      description: 'Construction date and the name of the designer.',
+      resolve() {
+        throw new Error('secretBackstory is secret.');
+      },
+    },
+    primaryFunction: {
+      type: GraphQLString,
+      description: 'The primary function of the droid.',
+    },
+  }),
+  interfaces: [characterInterface],
+});
+
+/**
+ * This is the type that will be the root of our query, and the
+ * entry point into our schema. It gives us the ability to fetch
+ * objects by their IDs, as well as to fetch the undisputed hero
+ * of the Star Wars trilogy, R2-D2, directly.
+ *
+ * This implements the following type system shorthand:
+ *   type Query {
+ *     hero(episode: Episode): Character
+ *     human(id: String!): Human
+ *     droid(id: String!): Droid
+ *   }
+ *
+ */
+const queryType = new GraphQLObjectType({
+  name: 'Query',
+  fields: () => ({
+    hero: {
+      type: characterInterface,
+      args: {
+        episode: {
+          description:
+            'If omitted, returns the hero of the whole saga. If ' +
+            'provided, returns the hero of that particular episode.',
+          type: episodeEnum,
+        },
+      },
+      resolve: (root, { episode }) => getHero(episode),
+    },
+    human: {
+      type: humanType,
+      args: {
+        id: {
+          description: 'id of the human',
+          type: new GraphQLNonNull(GraphQLString),
+        },
+      },
+      resolve: (root, { id }) => getHuman(id),
+    },
+    droid: {
+      type: droidType,
+      args: {
+        id: {
+          description: 'id of the droid',
+          type: new GraphQLNonNull(GraphQLString),
+        },
+      },
+      resolve: (root, { id }) => getDroid(id),
+    },
+  }),
+});
+
+/**
+ * Finally, we construct our schema (whose starting query type is the query
+ * type we defined above) and export it.
+ */
+export const StarWarsSchema = new GraphQLSchema({
+  query: queryType,
+  types: [humanType, droidType],
+  directives: [GraphQLDeferDirective],
+  // TODO: Eventually bake in @defer as a standard directive and do validation
```

Comment:
I don't think we should merge this is until that is done (baked in)

Reply:
Yep! Instead of modifying the validation phase of graphql.js, I think a more minimal approach would be to modify the call to GraphQLSchema's constructor within AS2, to pass in the additional schema directives.